core/src/main/java/net/sourceforge/jnlp/services/XBasicService.java

```diff
@@ -192,34 +187,30 @@ public boolean showDocument(URL url) {
 // ALWAYS-ASK, or directly via BROWSER of deployment.browser.path , it still should be better then it was
 // in all cases, the mime recognition is much harder then .jnlp suffix
 
-            String urls = url.toExternalForm();
+            final String urls = url.toExternalForm();
             LOG.debug("showDocument for: {}", urls);
 
-            DeploymentConfiguration config = JNLPRuntime.getConfiguration();
-            String command = config.getProperty(ConfigurationConstants.KEY_BROWSER_PATH);
+            final DeploymentConfiguration config = JNLPRuntime.getConfiguration();
+            final String command = config.getProperty(ConfigurationConstants.KEY_BROWSER_PATH);
             //for various debugging
             //command=DeploymentConfiguration.ALWAYS_ASK;
             if (command != null) {
                 LOG.debug("{} located. Using: {}", ConfigurationConstants.KEY_BROWSER_PATH, command);
                 return exec(command, urls);
             }
             if (System.getenv(ConfigurationConstants.BROWSER_ENV_VAR) != null) {
-                command = System.getenv(ConfigurationConstants.BROWSER_ENV_VAR);
+                final String cmd = System.getenv(ConfigurationConstants.BROWSER_ENV_VAR);
                 LOG.debug("variable {} located. Using: {}", ConfigurationConstants.BROWSER_ENV_VAR, command);
-                return exec(command, urls);
+                return exec(cmd, urls);
             }
+
             if (JNLPRuntime.isHeadless() || !Desktop.isDesktopSupported()) {
-                command = promptForCommand(urls, false);
-                return exec(command, urls);
+                final String cmd = promptForCommand(urls, false);
+                return exec(cmd, urls);
             } else {
-                if (Desktop.isDesktopSupported()) {
-                    LOG.debug("using default browser");
-                    Desktop.getDesktop().browse(url.toURI());
-                    return true;
-                } else {
-                    LOG.debug("dont know what to do");
-                    return false;
-                }
```

Comment:
There is change of behavior. Original codewas returning false, now it is throeing exception instead. How is it handled in consuming code?

Reply:
The else branch was unreachable. So in the inner if it would always take the if branch which is the one I preserved.
There code still returns a boolean and is not throwing an exception

```
if (a || !b) {
  // either a or not b is true
} else {
  // always not a and b
}

```

Reply:
I overlooked the first condition. TY!